src/core/lender/distributor/base.sol

```diff
@@ -47,39 +47,63 @@ contract Distributor is DSNote, Math {
     TrancheLike public senior;
     TrancheLike public junior;
 
-    constructor(address shelf_)  public {
+    constructor() public {
         wards[msg.sender] = 1;
-        shelf = ShelfLike(shelf_);
     }
 
     function depend (bytes32 what, address addr) public auth {
         if (what == "shelf") { shelf = ShelfLike(addr); }
-        if (what == "junior") { junior = TrancheLike(addr); }
+        else if (what == "junior") { junior = TrancheLike(addr); }
         else if (what == "senior") { senior = TrancheLike(addr); }
         else revert();
     }
 
+    /// moves balance from junior tranche to senior.
+    /// if senior tranche has a debt.
+    function _balanceTranches() internal {
+        if(address(senior) == address(0)) {
+            return;
+        }
+
+        uint seniorDebt = senior.debt();
+        uint juniorBalance = junior.balance();
+        if(juniorBalance > 0 && seniorDebt > 0) {
+            uint amount = seniorDebt;
+            if (amount > juniorBalance) {
+                amount = juniorBalance;
+            }
+            // move junior reserve to senior
+            senior.repay(address(junior), amount);
+        }
+    }
 
     /// handles requests from the shelf contract (borrower side)
     function balance() public {
+        _balanceTranches();
```

Comment:
I think It would also make sense to call borrowTranches() when investing new capital into junior.  Then we would always make sure to rebalance when new capital arrives